# Integrate genres into the main dataframe

In [1]:
import pandas as pd

Read in the clean dataframe

In [2]:
ratings = pd.read_csv('Ratings Clean.csv')

In [14]:
len(ratings.index)

85855

There are 85855 films in our dataset, each of which have one or more genres associated with them in the following format:  
  
0                          Romance  
1          Biography, Crime, Drama  
2                            Drama  
3                   Drama, History  
4        Adventure, Drama, Fantasy  
                   ...              
85850                       Comedy  
85851                Comedy, Drama  
85852                        Drama  
85853                Drama, Family  
85854                        Drama  
  
We see that some films have multiple genres in the same string, so we have some cleaning to do. We need to figure out a way to find all the unique genres and create columns for each one, indicating which genre is associated with a given film using a 1 (True) or 0 (False).

Start by creating a list of the genres for each movie.

In [7]:
genres = pd.read_csv('IMDb movies.csv').genre

The following code iterates through the list of movie's genres and, if there are multiple, splits them into lists. If we come across a genre that we haven't seen yet, we add it to the list ```genres_unique```.

In [27]:
genres_unique = []

for film in range(0,len(genres)):
    film_genres = genres[film].split(", ")
    for genre in film_genres:
        if (genre not in genres_unique):
            genres_unique.append(genre)

Let's check if the list is indeed unique:

In [41]:
#pd.Series(genres_unique).nunique()
# 25
#len(genres_unique)
# 25

25

There are 25 unique genres associated with the films in the dataset.